### DISCUSSION TITLE
应用备份恢复失败

### DISCUSSION THREAD

**Discussion by rf11291** [1Panel-dev/1Panel#6259] [Q&A]
### 1Panel 版本

v1.10.13-lts

### 问题描述

将R2的备份应用文件下载到本地后再上传至服务器，然后进行恢复会出现`服务内部错误: the wrong recovery package does not have app.json or app.tar.gz files`这个提示，但是我去目录中检查发现，压缩文件已经被解压成功了，文件夹里也有app.json，app.tar.gz这两个文件，但就是一直提示失败

### 重现步骤

在其他服务器的面板上操作亦是如此

### 期待的正确结果

如暂时无法解决便请告诉我如何手动恢复应用，例如`openresty`，谢谢！

**Comment by rf11291** [comment]
![PixPin_2024-08-27_03-08-43](https://github.com/user-attachments/assets/4c836349-68ba-440f-b118-f0ad3934297f)

**Comment by zhengkunwang223** [comment]
OpenResty 恢复，网站列表中不会出现网站，只是把文件恢复了 这个你要先知道
你先手动解压你的备份文件 然后截图看一下都有什么

**Reply by moxia** [reply]
OpenResty 恢复，网站列表中不会出现网站，只是把文件恢复了
-----
这一点也太坑了，如何备份恢复网站列表？

**Comment by rf11291** [comment]
> OpenResty 恢复，网站列表中不会出现网站，只是把文件恢复了 这个你要先知道 你先手动解压你的备份文件 然后截图看一下都有什么

解压后两个文件，我将app.tar.gz压缩文件解压覆盖了openresty的应用目录,但是怎么将app.json中的应用配置参数恢复？
![PixPin_2024-08-27_10-22-23](https://github.com/user-attachments/assets/48b212af-f877-4e97-9a2b-4196b3b724ee)
![PixPin_2024-08-27_10-26-44](https://github.com/user-attachments/assets/c40e8c7c-1a3d-4cb4-8d89-d5da5cb53369)

**Comment by zhengkunwang223** [comment]
> OpenResty 恢复，网站列表中不会出现网站，只是把文件恢复了 这个你要先知道 你先手动解压你的备份文件 然后截图看一下都有什么
> > 解压后两个文件，我将app.tar.gz压缩文件解压覆盖了openresty的应用目录,但是怎么将app.json中的应用配置参数恢复？


应用的目录在 app.tar.gz 中 但是你手动操作可能会带来问题 需要自行处理

**Comment by rf11291** [comment]
>  OpenResty 恢复，网站列表中不会出现网站，只是把文件恢复了 这个你要先知道 你先手动解压你的备份文件 然后截图看一下都有什么 
>> 解压后两个文件，我将app.tar.gz压缩文件解压覆盖了openresty的应用目录，但是怎么将app.json中的配置参数恢复?
>>> 应用的目录在 app.tar.gz 中 但是你手动操作可能会带来问题 需要自行处理

现在就是将应用目录的文件恢复了，但是现在怎么从app.json恢复应用的配置参数等，不然就会出现下面这张图的情况

![PixPin_2024-08-27_10-33-29](https://github.com/user-attachments/assets/01802e80-88a7-40ab-b53b-c8686d60f931)

**Reply by wanghe-fit2cloud** [reply]
目前版本支持同一服务器上的同一应用备份和恢复操作，但不支持跨服务器或跨 1Panel 的场景。
若需实现迁移效果，建议使用面板设置页面提供的快照功能，该功能将默认备份网站、已安装应用等内容。

**Comment by rf11291** [comment]
我目前的方法手动一个个创建...

moxia ***@***.***> 于 2025年1月20日周一 14:02写道：
